$NetBSD: patch-ac,v 1.3 2009/04/11 14:04:16 taca Exp $

Detect more header files: machine/byte_swap.h and net/pf/pfvar.h.
Prevent optimization with GCC 2.95.[123].

--- configure.orig	2008-12-23 13:19:34.000000000 +0900
+++ configure
@@ -25026,6 +25026,7 @@ for ac_header in \
 	ipl.h \
 	libc.h \
 	limits.h \
+  	machine/byte_swap.h \
 	malloc.h \
 	math.h \
 	memory.h \
@@ -25348,6 +25349,7 @@ done
 
 for ac_header in \
 	net/if.h \
+	net/pf/pfvar.h \
 	net/pfvar.h \
 	netinet/if_ether.h\
 	netinet/ip.h\
@@ -44563,8 +44565,8 @@ if test "$GCC" = "yes"; then
 	GCCVER=`$CC -v 2>&1 | awk '$2 ==  "version" {print $3}'`
 	case "$GCCVER" in
 	2.95.[123])
-		echo "Removing -O for gcc on $host with GCC $GCCVER"
-		CFLAGS="`echo $CFLAGS | sed -e 's/-O[0-9]*//'`"
+		echo "Making -O\[[2-9\]] to -O for gcc on $host with GCC $GCCVER"
+		CFLAGS="`echo $CFLAGS | sed -e 's/-O[[0-9]]*/-O/'`"
 		;;
 	esac
 fi
@@ -45894,7 +45896,7 @@ if test "$PF_TRANSPARENT" ; then
     { $as_echo "$as_me:$LINENO: checking if PF header file is installed" >&5
 $as_echo_n "checking if PF header file is installed... " >&6; }
     # hold on to your hats...
-    if test "$ac_cv_header_net_pfvar_h" = "yes"; then
+    if test "$ac_cv_header_net_pfvar_h" = "yes" || test "$ac_cv_header_net_pf_pfvar_h" = "yes"; then
         PF_TRANSPARENT="yes"
         cat >>confdefs.h <<\_ACEOF
 #define PF_TRANSPARENT 1
